internal/configs/version1/nginx-plus.ingress.tmpl

```diff
@@ -1,7 +1,7 @@
 # configuration for {{.Ingress.Namespace}}/{{.Ingress.Name}}
 {{range $upstream := .Upstreams}}
 upstream {{$upstream.Name}} {
-	zone {{$upstream.Name}} 256k;
+	{{if ne $upstream.UpstreamZoneSize "0"}}zone {{$upstream.Name}} {{$upstream.UpstreamZoneSize}}; {{else}} zone {{$upstream.Name}} 256k;{{end}}
```

Comment:
this can be simplified:
```suggestion
	zone {{$upstream.Name}} {{if ne $upstream.UpstreamZoneSize "0"}}{{$upstream.UpstreamZoneSize}}{{else}}256k{{end}};
```
You can use this simplification in the config template for virtual server as well.

Note: the change wasn't tested.